### PR TITLE
[ENG-217] Create store for turnkey users

### PR DESCRIPTION
**Pull request by Kefancao**
### Changelist
[Describe or list the changes made in this PR]

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

* **New Features**
  * Added support for managing TurnkeyUser records, including creation, upserting, and retrieval by various identifiers (EVM address, SVM address, email, suborg ID).
  * Introduced a new database table for TurnkeyUser data with optimized indexes for faster lookups.
  * Expanded query options to include additional fields and improved type safety for TurnkeyUser operations.

* **Tests**
  * Added comprehensive tests covering creation, upserting, and retrieval of TurnkeyUser records, including edge cases.

### CODE REVIEW

File: indexer/packages/postgres/src/db/migrations/migration_files/20250709170936_create_turnkey_users.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+export async function up(knex: Knex): Promise<void> {
+  return knex
+    .schema
+    .createTable('turnkey_users', (table) => {
+      table.text('suborg_id').primary(); // Primary key
+      table.text('username').nullable(); // optional
+      table.text('email').nullable().unique(); // optional but unique
+      table.text('svm_address').notNullable(); // indexed
+      table.text('evm_address').notNullable(); // indexed
```
Comment (UnbornAztecKing) [inline indexer/packages/postgres/src/db/migrations/migration_files/20250709170936_create_turnkey_users.ts]: why aren't the addresses unique?

File: indexer/packages/postgres/src/stores/turnkey-users-table.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  evmAddress: string,
+  options: Options = DEFAULT_POSTGRES_OPTIONS,
+): Promise<TurnkeyUserFromDatabase | undefined> {
+  const baseQuery: QueryBuilder<TurnkeyUserModel> = setupBaseQuery<TurnkeyUserModel>(
+    TurnkeyUserModel,
+    options,
+  );
+  return baseQuery
+    .where(TurnkeyUserColumns.evm_address, evmAddress)
+    .first();
```
Comment (UnbornAztecKing) [inline indexer/packages/postgres/src/stores/turnkey-users-table.ts:24]: DB schema doesn't constrain the table such that only one of these may be returned.

File: indexer/packages/postgres/src/types/turnkey-user-types.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/* ------- TURNKEY USER TYPES ------- */
```
Comment (UnbornAztecKing) [inline indexer/packages/postgres/src/types/turnkey-user-types.ts]: please don't add these in any form, it's just noise.

every type you added has TurnkeyUser in the name and the module is named turnkey-user-types, which is also the pattern elsewhere this has been done in this repo, and all should be cleaned up.

File: indexer/packages/postgres/src/types/db-model-types.ts [hunk 4]
```diff
@@ -336,6 +336,17 @@ export interface FundingPaymentsFromDatabase {
   fundingIndex: string,
 }
 
+export interface TurnkeyUserFromDatabase {
+  suborg_id: string,
+  username?: string,
```
Comment (shrenujb) [inline indexer/packages/postgres/src/types/db-model-types.ts:341]: dont need this currently as its not going to be used
